CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${BINARY_OUT_DIR})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${BINARY_OUT_DIR})
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${BINARY_OUT_DIR})
+
+foreach(OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES})
+	string(TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG)
+	set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${BINARY_OUT_DIR})
+	set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${BINARY_OUT_DIR})
+	set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${BINARY_OUT_DIR})
+endforeach(OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES)
+
+link_directories(${CMAKE_CURRENT_BINARY_DIR}/lib) # For future use with testing library.
+
+##
+# Project
+##
+file(GLOB_RECURSE INCLUDES "${SG14_INCLUDE_DIRECTORY}/*.hpp" "${SG14_INCLUDE_DIRECTORY}/*.h" "${SG14_INCLUDE_DIRECTORY}/*.tpp")
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE ${SG14_INCLUDE_DIRECTORY})
+target_sources(${PROJECT_NAME} INTERFACE
+	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+	$<BUILD_INTERFACE:${INCLUDES}>
+)
+target_include_directories(${PROJECT_NAME} INTERFACE
+	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+	$<BUILD_INTERFACE:${SG14_INCLUDE_DIRECTORY}>
+)
+
+##
+# Unit Tests
+##
+set(TEST_SOURCE_FILES
+    ${SG14_TEST_SOURCE_DIRECTORY}/main.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/inplace_function_test.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/ring_test.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/slot_map_test.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/transcode_test.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/uninitialized_test.cpp
+    ${SG14_TEST_SOURCE_DIRECTORY}/unstable_remove_test.cpp
+)
+
+set(TEST_NAME ${PROJECT_NAME}_tests)
+add_executable(${TEST_NAME} ${TEST_SOURCE_FILES})
+target_link_libraries(${TEST_NAME} ${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT})
+target_include_directories(${TEST_NAME} PRIVATE "${SG14_TEST_SOURCE_DIRECTORY}")
+
+# Compile options
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+	target_compile_options(${TEST_NAME} PRIVATE -Wall -Wextra -Wfatal-errors -Werror)
+	set_source_files_properties(${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp PROPERTIES
+		COMPILE_FLAGS "-Wno-unused-parameter"
+	)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+	target_compile_options(${TEST_NAME} PRIVATE -Wall -Wextra -Wfatal-errors -Werror)
+	set_source_files_properties(${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp PROPERTIES
+		COMPILE_FLAGS "-Wno-unused-parameter"
+	)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+  # using Intel C++
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+	set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT ${TEST_NAME})
+	target_compile_options(${TEST_NAME} PRIVATE /Zc:__cplusplus /permissive- /W4)# /WX) temporarily allow warnings
+	add_definitions(-DNOMINMAX -D_SCL_SECURE_NO_WARNINGS)
+	set_source_files_properties(${SG14_TEST_SOURCE_DIRECTORY}/plf_colony_test.cpp PROPERTIES
+		COMPILE_FLAGS "/wd4100 /wd4456"
```

Comment:
Likewise it would help to explain in the commit message what are these diagnostic numbers you're suppressing. Cf. https://github.com/zenorogue/hyperrogue/commit/51f59d7bcc919bf8f0a3fbdfc45a228b4a77b6f2

Reply:
These are ported from the old cmake file. You would have to ask colony author for the specifics.